WW3 changes by @awest-noaa

zacharyburnettNOAA opened the pull request.

awest-noaa commented:
Zach, before moving ahead with this, Takis still needs to make one correction to the ADCIRC submodule file adc_cap.F90 he and I discussed on Monday. Please check with him

zacharyburnettNOAA commented:
I have rebased this branch on top of the changes made in the meantime to `develop`, so that these commits are linear and put on top (chronologically) and there are no merge conflicts. Additionally, I updated the ADCIRC submodule to point to the new NUOPC changes in the `notre_dame` branch that we recently rebased.

awest-noaa commented:
@zacharyburnettNOAA thanks for these changes. As discussed yesterday, these can be tested using the following (offline) set of regtests on Hera: /scratch2/COASTAL/coastal/save/Andre.VanderWesthuysen/OCS-NEMS3/CoastalApp/regtests
